### PR TITLE
fibonacci

### PR DESCRIPTION
**Pull request by DanielZerihon**

### CODE REVIEW

File: Basic/fibonacci/fibonacci.py [hunk 1]
```diff
@@ -1,2 +1,24 @@
...
+        if(idx == 1): 
+            return 1
+        if(idx == 2): 
+            return 1
+
+    i = 2 
+    firstDigit = 1 
+    secondDigit = 2
+    tempDigit = 0
+    while(i < idx):
```
Comment (talfridmen) [inline Basic/fibonacci/fibonacci.py:16]: why while loop and not for loop?

File: Basic/fibonacci/fibonacci.py [hunk 2]
```diff
@@ -1,2 +1,24 @@
...
-    pass # Implement Here!
\ No newline at end of file
+    if(idx <= 2):
+        if(idx < 0): 
+            return -1
+        if(idx == 0): 
+            return 0
+        if(idx == 1): 
+            return 1
+        if(idx == 2):
```
Comment (talfridmen) [inline Basic/fibonacci/fibonacci.py:9]: why specifically so many ifs, and not do it as part of the main loop?

File: Basic/fibonacci/fibonacci.py [hunk 3]
```diff
@@ -1,2 +1,24 @@
...
+            return -1
+        if(idx == 0): 
+            return 0
+        if(idx == 1): 
+            return 1
+        if(idx == 2): 
+            return 1
+
+    i = 2 
+    firstDigit = 1
```
Comment (talfridmen) [inline Basic/fibonacci/fibonacci.py:13]: Digit is not the right word. Digit is just one character. You want Number here instead :)

File: Basic/fibonacci/fibonacci.py [hunk 4]
```diff
@@ -1,2 +1,24 @@
...
+    i = 2 
+    firstDigit = 1 
+    secondDigit = 2
+    tempDigit = 0
+    while(i < idx):
+        tempDigit = secondDigit
+        secondDigit += firstDigit
+        firstDigit = tempDigit
+        i = i + 1 
+    print(f"fibonacci[{idx}] = {firstDigit}")
```
Comment (talfridmen) [inline Basic/fibonacci/fibonacci.py:21]: why are you printing stuff?

File: Basic/fibonacci/fibonacci.py [hunk 5]
```diff
@@ -1,2 +1,24 @@
...
+    return firstDigit
+
+fibonacci(10)
```
Comment (talfridmen) [inline Basic/fibonacci/fibonacci.py:24]: What is this?
leave a clean code - only the required implementations

**Review by talfridmen — COMMENTED** [review]
see comments bellow

File: Basic/fibonacci/fibonacci.py (hunk 4 above)
Comment (DanielZerihon) [reply Basic/fibonacci/fibonacci.py:21]: removed

File: Basic/fibonacci/fibonacci.py (hunk 5 above)
Comment (DanielZerihon) [reply Basic/fibonacci/fibonacci.py:24]: removed

File: Basic/fibonacci/fibonacci.py (hunk 2 above)
Comment (DanielZerihon) [reply Basic/fibonacci/fibonacci.py:9]: fixed

**Review by talfridmen — APPROVED** [review]
Good job :) Looks much better

**Pull request merged**